app/src/main/java/com/google/samples/apps/sunflower/di/DatabaseModule.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.di
+
+import android.content.Context
+import com.google.samples.apps.sunflower.data.AppDatabase
+import com.google.samples.apps.sunflower.data.GardenPlantingDao
+import com.google.samples.apps.sunflower.data.PlantDao
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ApplicationComponent
+import dagger.hilt.android.qualifiers.ApplicationContext
+
+@InstallIn(ApplicationComponent::class)
+@Module
+class DatabaseModule {
+
+    @Provides
+    fun provideAppDatabase(@ApplicationContext context: Context): AppDatabase {
```

Comment:
As this is accessed a lot of times throughout the app and might be expensive in some devices. Let's annotate as `@Singleton` so that we always reuse the same instances. 

The daos don't need to as they're quick to retrieve